cve_bin_tool/output_engine.py

```diff
@@ -0,0 +1,39 @@
+import csv
+
+from datetime import datetime
+from builtins import hash
+
+
+class OutputEngine:
+    def __init__(self, filename=None, modules=None):
+        self.filename = filename
+        self.modules = modules
+
+        if self.filename == None:
+            now = datetime.now()
+            self.filename = str(hash(now)).strip("-")
+
+    def output_cves(self, outfile):
+
+        """ Output a list of CVEs
+        format is modules[checker_name][version] = dict{id: severity}
+        """
+        writer = csv.writer(outfile)
+        for modulename, versions in self.modules.items():
+            for version, cve_list in versions.items():
+                for cve_number, cve_severity in cve_list.items():
+                    row = [modulename, version, cve_number, cve_severity]
+                    writer.writerow(row)
+
+    def output_csv(self):
+
+        """ Generate a CSV file for list of CVE """
+
+        outfile = open(f"Output/{self.filename}.csv", "a")
```

Comment:
Add a log call here to tell the user what the filename was in case it was auto-generated.  You can set something in `__init__` so you know when you generated one and only print it there.  Or equally, you could move the code out of `__init__` and into this function (which might make more sense).

And as I said above, I think we should skip the directory and just make a bare file.  

We might need a check in case the user specifies a file that they can't write to/directory that doesn't exist here.  Python's error should be pretty obvious if it can't open the file, but if we have a check then we can revert to console output or just terminate more nicely.